CSPAgent.py

```diff
@@ -1,42 +1,126 @@
 from game import *
+	
+	#CSPAgent focuses on 'observe_suggestion()' function.
+	#- It observes the exchange as first opponent is the suggester
+	
 
-class SimpleAgent(Agent):
+class CSPAgent(Agent):
+	def __init__(self, name):
+		super().__init__(name)
+
+		#Make opponent hands
+		self.first_opponent_hand = Hand()
+		self.second_opponent_hand = Hand()
+
+		print("START")
+		print("cards: {}".format(self.first_opponent_hand.get_cards()))
+		print("cards: {}".format(self.second_opponent_hand.get_cards()))
+
+		#Sets to keep track of potential domains for each hand
+		self.first_opponent_sets = []
+		self.second_opponent_sets = []
+
+		#Keep track of prev suggestion
+		self.past_suggestion = [None]*3
 
 	def make_move(self):
 		'''
 		decide when to make a suggestion or accusation
 		suggestion is of type Suggestion
 		accusation is a dict where key is the type, and the value is the card
+
+		Update player
+		Update casefile
+		Make guess based on values not in sets
 		'''
-		weapon_dom = np.shuffle(self.caseFileWeapon.cur_dom())
-		room_dom = np.shuffle(self.caseFileRoom.cur_dom())
-		suspect_dom = np.shuffle(self.caseFileRoom.cur_dom())
+		weapon_dom = self.caseFileWeapon.cur_domain()
+		room_dom = self.caseFileRoom.cur_domain()
+		suspect_dom = self.caseFileSuspect.cur_domain()
+
+		#For the first tern - randomly make a suggestion
+		if None in self.past_suggestion:
+			#Prune agents cards from opponents' hands
+			self.first_opponent_hand.pruneHand(self.hand)
+			self.second_opponent_hand.pruneHand(self.hand)
+
+			np.random.shuffle(weapon_dom)
+			np.random.shuffle(room_dom)
+			np.random.shuffle(suspect_dom)
+
+			self.past_suggestion[0] = weapon_dom[0]
+			self.past_suggestion[1] = room_dom[0]
+			self.past_suggestion[2] = suspect_dom[0]
+
+			suggestion = Suggestion(self.name, self.firstOppName, weapon_dom[0], room_dom[0], suspect_dom[0])
+			return suggestion
+
+		#Update the player's hands and sets
+		self._update_player(self.first_opponent_hand, self.first_opponent_sets)
+		self._update_player(self.second_opponent_hand, self.second_opponent_sets)
+		
+		#Prune assigned values in one hand from the other
+		self._update_each_other()
+
+		#Prune assigned values from the hands from the cur dom of the casefile
+		self._update_case(self.first_opponent_hand)
+		self._update_case(self.second_opponent_hand)
+
+		print("Player {}'s cards: {}".format(self.firstOppName, self.first_opponent_hand.get_cards()))
```

Comment:
Kinda want to keep them in case something breaks / someone wants to reuse elements (for now!)